src/pepr/operator/controllers/network/authorizationPolicies.spec.ts

```diff
@@ -1112,3 +1114,125 @@ describe("createDenyAllExceptWaypointPolicy", () => {
     ]);
   });
 });
+
+describe("findMatchingSsoClient", () => {
+  test("returns undefined when selector is undefined", () => {
+    const pkg: UDSPackage = {
+      metadata: { name: "app", namespace: "ns" },
+      spec: {
+        sso: [{ clientId: "c1", name: "", enableAuthserviceSelector: { app: "a" } }],
+        network: { serviceMesh: { mode: Mode.Ambient } },
+      },
+    };
+
+    expect(
+      findMatchingSsoClient(pkg, undefined as unknown as Record<string, string>),
+    ).toBeUndefined();
+  });
+
+  test("returns undefined when mesh mode is not Ambient", () => {
+    const pkg: UDSPackage = {
+      metadata: { name: "app", namespace: "ns" },
+      spec: {
+        sso: [{ clientId: "c1", name: "", enableAuthserviceSelector: { app: "a" } }],
+        network: { serviceMesh: { mode: Mode.Sidecar } },
+      },
+    };
+
+    expect(findMatchingSsoClient(pkg, { app: "a" })).toBeUndefined();
+  });
+
+  test("prefilters to enabled clients only (missing/null/undefined selector are ignored)", () => {
+    const pkg: UDSPackage = {
+      metadata: { name: "app", namespace: "ns" },
+      spec: {
+        sso: [
+          { clientId: "d1", name: "" },
+          { clientId: "d3", name: "", enableAuthserviceSelector: undefined },
+        ],
+        network: { serviceMesh: { mode: Mode.Ambient } },
+      },
+    };
+
+    expect(findMatchingSsoClient(pkg, { app: "anything" })).toBeUndefined();
+  });
+
+  test("empty SSO selector ({}) matches any provided selector (namespace-wide)", () => {
+    const pkg: UDSPackage = {
+      metadata: { name: "app", namespace: "ns" },
+      spec: {
+        sso: [{ clientId: "c1", name: "", enableAuthserviceSelector: {} }],
+        network: { serviceMesh: { mode: Mode.Ambient } },
+      },
+    };
+
+    expect(findMatchingSsoClient(pkg, {})).toMatchObject({ clientId: "c1" });
+    expect(findMatchingSsoClient(pkg, { app: "x" })).toMatchObject({ clientId: "c1" });
+  });
+
+  test("non-empty SSO selector uses any-label matching semantics", () => {
+    const pkg: UDSPackage = {
+      metadata: { name: "app", namespace: "ns" },
+      spec: {
+        sso: [{ clientId: "c1", name: "", enableAuthserviceSelector: { app: "a", tier: "prod" } }],
+        network: { serviceMesh: { mode: Mode.Ambient } },
+      },
+    };
+
+    // Matches because 'tier' label matches (any-label semantics)
+    expect(findMatchingSsoClient(pkg, { tier: "prod" })).toMatchObject({ clientId: "c1" });
```

Comment:
This test is the one that made me suspicious of the logic I commented on (and this test would need to be changed if that tracks).